src/test/regress/expected/multi_insert_select_conflict.out

```diff
@@ -0,0 +1,423 @@
+ col_1 | col_2 
+-------+-------
+     1 |     1
+     2 |     2
+     3 |     3
+     4 |     4
+     5 |     5
+(5 rows)
+
+-- Test with multiple subqueries. Query is wrapped by CTE to return ordered result.
+WITH inserted_table AS (
+	INSERT INTO target_table
+	SELECT 
+		col_1, col_2
+	FROM (
+		(SELECT 
+			col_1, col_2, col_3 
+		FROM
+			source_table_1
+		LIMIT 5)
+		UNION
+		(SELECT
+			col_1, col_2, col_3
+		FROM
+			source_table_2
+		LIMIT 5)
+	) as foo
+	ON CONFLICT(col_1) DO UPDATE SET col_2 = 0 RETURNING *
+) SELECT * FROM inserted_table ORDER BY 1;
+DEBUG:  data-modifying statements are not supported in the WITH clauses of distributed queries
+DEBUG:  generating subplan 18_1 for CTE inserted_table: INSERT INTO on_conflict.target_table (col_1, col_2) SELECT col_1, col_2 FROM ((SELECT source_table_1.col_1, source_table_1.col_2, source_table_1.col_3 FROM on_conflict.source_table_1 LIMIT 5) UNION (SELECT source_table_2.col_1, source_table_2.col_2, source_table_2.col_3 FROM on_conflict.source_table_2 LIMIT 5)) foo ON CONFLICT(col_1) DO UPDATE SET col_2 = 0 RETURNING target_table.col_1, target_table.col_2
+DEBUG:  Set operations are not allowed in distributed INSERT ... SELECT queries
+DEBUG:  Plan 18 query after replacing subqueries and CTEs: SELECT col_1, col_2 FROM (SELECT intermediate_result.col_1, intermediate_result.col_2 FROM read_intermediate_result('18_1'::text, 'binary'::citus_copy_format) intermediate_result(col_1 integer, col_2 integer)) inserted_table ORDER BY col_1
+DEBUG:  Collecting INSERT ... SELECT results on coordinator
+DEBUG:  push down of limit count: 5
+DEBUG:  generating subplan 20_1 for subquery SELECT col_1, col_2, col_3 FROM on_conflict.source_table_1 LIMIT 5
+DEBUG:  push down of limit count: 5
+DEBUG:  generating subplan 20_2 for subquery SELECT col_1, col_2, col_3 FROM on_conflict.source_table_2 LIMIT 5
+DEBUG:  generating subplan 20_3 for subquery SELECT intermediate_result.col_1, intermediate_result.col_2, intermediate_result.col_3 FROM read_intermediate_result('20_1'::text, 'binary'::citus_copy_format) intermediate_result(col_1 integer, col_2 integer, col_3 integer) UNION SELECT intermediate_result.col_1, intermediate_result.col_2, intermediate_result.col_3 FROM read_intermediate_result('20_2'::text, 'binary'::citus_copy_format) intermediate_result(col_1 integer, col_2 integer, col_3 integer)
+DEBUG:  Plan 20 query after replacing subqueries and CTEs: SELECT col_1, col_2 FROM (SELECT intermediate_result.col_1, intermediate_result.col_2, intermediate_result.col_3 FROM read_intermediate_result('20_3'::text, 'binary'::citus_copy_format) intermediate_result(col_1 integer, col_2 integer, col_3 integer)) foo
+ col_1 | col_2 
+-------+-------
+     1 |     0
+     2 |     0
+     3 |     0
+     4 |     0
+     5 |     0
+     6 |     0
+     7 |     0
+     8 |     0
+     9 |     0
+    10 |     0
+(10 rows)
+
+-- Get the select part from cte and do nothing on conflict
```

Comment:
I've slightly changed the query, and it didn't play nice. Do you know why is this happening? Or is that expected?

```SQL
WITH cte AS
  (SELECT col_1, col_2
   FROM source_table_1)
INSERT INTO target_table
SELECT source_table_1.col_1,
       source_table_1.col_2
FROM cte,
     source_table_1
WHERE cte.col_1 = source_table_1.col_1 ON CONFLICT DO NOTHING;
ERROR:  cannot perform distributed planning for the given modification
DETAIL:  Select query cannot be pushed down to the worker.
```